### PR TITLE
Bug/support tls service monitors

### LINKED ISSUE #215: Chart is missing support for pulling metrics with TLS
I enabled TLS in Cortex, the service monitors also need the ability to pull metrics over TLS.

### PR DESCRIPTION
**Pull request by danfromtitan**
**What this PR does**:
Add TLS configuration to ServiceMonitors templates.

**Which issue(s) this PR fixes**:
Fixes #215

**Checklist**
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 * [ENHANCEMENT] Use FQDN for memcached addresses #175
 * [ENHANCEMENT] Optionally generate endpoints for `X-Scope-OrgID` injection (multi-tenancy) #180
 * [ENHANCEMENT] Optionally configure Basic Auth within Nginx #205
+* [ENHANCEMENT] Add service monitors TLS support #215
```
Comment (ShuzZzle) [inline CHANGELOG.md:16]:
```suggestion
* [ENHANCEMENT] Add service monitors TLS support #216
```

wrong PR number

**Comment by ShuzZzle** [comment]
Hi @danfromtitan,

Just a heads up. We have not forgot about your PR but considering a more dynamic approach to extending the ServiceMonitor options since there are so so many different options.

Maybe consider having something like this
```yaml
{{- with .Values.ingester.serviceMonitor.extraEndpointSpec }}
{{ tpl . $ | nindent <LEVEL> }}
{{- end }}
```
```yaml
# values.yml
    extraEndpointSpec:
      scheme: HTTPS
      tlsConfig: foo
```

maybe even for the whole [ServiceMonitorSpec](https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#servicemonitorspec) since there are so many different options

**Comment by danfromtitan** [comment]
I can see the value of adding a generic extraEndpointSpec, but taking it at the ServiceMonitorSpec level it too much IMO. Not only the configuration becomes hard to figure out for a newcomer, but you also have different selector.matchLabels for each component that you'd want to feed from the template.

**Comment by danfromtitan** [comment]
@ShuzZzle, I added extraEndpointSpec as discussed above.

**Review by kd7lxl — APPROVED** [review]

**Review by ShuzZzle — APPROVED** [review]

**Pull request merged**